nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-processors/src/main/java/com/marklogic/nifi/processor/PutMarkLogic.java

```diff
@@ -0,0 +1,382 @@
+            " based on the URI")
+        .allowableValues(Format.JSON.name(), Format.XML.name(), Format.TEXT.name(), Format.BINARY.name(), Format.UNKNOWN.name())
+        .addValidator(NO_VALIDATION_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor JOB_ID = new PropertyDescriptor.Builder()
+        .name("Job ID")
+        .displayName("Job ID")
+        .description("ID for the WriteBatcher job")
+        .addValidator(NO_VALIDATION_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor JOB_NAME = new PropertyDescriptor.Builder()
+        .name("Job Name")
+        .displayName("Job Name")
+        .description("Name for the WriteBatcher job")
+        .addValidator(NO_VALIDATION_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor MIMETYPE = new PropertyDescriptor.Builder()
+        .name("MIME type")
+        .displayName("MIME type")
+        .description("MIME type for each document; if not specified, MarkLogic will determine the " +
+            "MIME type based on the URI")
+        .addValidator(NO_VALIDATION_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor PERMISSIONS = new PropertyDescriptor.Builder()
+        .name("Permissions")
+        .displayName("Permissions")
+        .defaultValue("rest-reader,read,rest-writer,update")
+        .description("Comma-delimited sequence of permissions - role1, capability1, role2, " +
+            "capability2 - to add to each document")
+        .addValidator(NO_VALIDATION_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor TEMPORAL_COLLECTION = new PropertyDescriptor.Builder()
+        .name("Temporal collection")
+        .displayName("Temporal collection")
+        .description("The temporal collection to use for a temporal document insert")
+        .addValidator(NO_VALIDATION_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor TRANSFORM = new PropertyDescriptor.Builder()
+        .name("Server transform")
+        .displayName("Server transform")
+        .description("(Optional) The name of REST server transform to apply to every document as it's" +
+            " written")
+        .addValidator(NO_VALIDATION_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor URI_ATTRIBUTE_NAME = new PropertyDescriptor.Builder()
+        .name("URI attribute name")
+        .displayName("URI attribute name")
+        .defaultValue("uuid")
+        .required(true)
+        .description("The name of the FlowFile attribute whose value will be used as the URI")
+        .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor URI_PREFIX = new PropertyDescriptor.Builder()
+        .name("URI prefix")
+        .displayName("URI prefix")
+        .description("(Optional) The prefix to prepend to each URI")
```

Comment:
Remove `(Optional)` and replace with an explicit `required(boolean)` declaration.